src/pages/OnboardingPage/steps/InputBirthdayStep.tsx

```diff
@@ -1,18 +1,103 @@
+import {
+  useForm,
+  type SubmitHandler,
+  type SubmitErrorHandler,
+} from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { z } from 'zod';
 import { css } from '@emotion/react';
+import { useMutation } from '@tanstack/react-query';
 import Button from '@/components/Button';
 import { useFunnelContext } from '@/contexts/useFunnelContext';
 import textStyles from '@/styles/textStyles';
+import memberAPI from '@/api/member/apis';
+import LoadingSpinner from '@/components/LoadingSpinner';
 import StepTemplate from '../components/StepTemplate';
 import onboardingStyles from '../styles';
 
+const formLiteral = {
+  year: {
+    min: 1800,
+    max: 9999,
+    length: 4,
+  },
+  month: {
+    min: 1,
+    max: 12,
+    length: 2,
+  },
+  day: {
+    min: 1,
+    max: 31,
+    length: 2,
+  },
+} as const;
+
+const L = formLiteral;
+
+const formSchema = z.object({
+  year: z.coerce.number().int().min(L.year.min).max(L.year.max),
+  month: z.coerce.number().int().min(L.month.min).max(L.month.max),
+  day: z.coerce.number().int().min(L.day.min).max(L.day.max),
+});
+
+type Inputs = z.infer<typeof formSchema>;
+
 const InputBirthdayStep = () => {
+  const { register, handleSubmit } = useForm<Inputs>({
+    resolver: zodResolver(formSchema),
+  });
+
   const { toNext } = useFunnelContext();
+  const { mutateAsync, isPending } = useMutation({
+    mutationFn: memberAPI.patchBirthday,
+  });
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const name = e.target.name;
+    const value = e.target.value.replace(/[^0-9]/g, '');
+
+    if (!(name === 'day' || name === 'month' || name === 'year')) {
+      return;
+    }
```

Comment:
p5; 이거 이렇게 표현할 수 있지 않을까요?
```suggestion
    if (!['day', 'month', 'year'].includes(name)) {
      return;
    }
```

Reply:
`.includes` 가 런타임에서 동작하는 함수이다보니.. 타입스크립트가 정적으로 타입 추론을 못 해 주더라구요 😂

![image](https://github.com/dnd-side-project/dnd-10th-4-frontend/assets/50488780/d04be945-1aec-475c-89d8-3ef7c367f95c)

Reply:
아하.. `.includes` 를 사용하려면 `['day', 'month', 'year']` 이 배열의 타입을 정의해줘야 사용할 수 있겠네요 🥲

원래 코드로 하는 게 좋을 것 같습니다!